components/tidb_query/src/codec/convert.rs

```diff
@@ -757,11 +757,8 @@ impl ConvertTo<f64> for &[u8] {
                 }
                 Ok(val)
             }
-            // if reaches here, it means our code has bug
-            Err(err) => {
-                debug_assert!(false);
-                Err(box_err!("parse float err: {}, this is a bug", err))
-            }
+            // if reaches here, it means our code has bug.
```

Comment:
What about using `unreachable` here and remove this comment.

Reply:
Unreachable is not a good idea, it's not an unrecoverable bug (In principle we should be able to parse into float number because of the string comes from `get_valid_float_prefix`).

Reply:
This PR has fixed a bug of `get_valid_float_prefix`.